axum/src/routing/tests/mod.rs

```diff
@@ -316,33 +316,29 @@ async fn middleware_applies_to_routes_above() {
 }
 
 #[tokio::test]
-async fn with_trailing_slash() {
+async fn not_found_for_extra_trailing_slash() {
     let app = Router::new().route("/foo", get(|| async {}));
 
     let client = TestClient::new(app);
 
     let res = client.get("/foo/").send().await;
-    assert_eq!(res.status(), StatusCode::PERMANENT_REDIRECT);
-    assert_eq!(res.headers().get("location").unwrap(), "/foo");
+    assert_eq!(res.status(), StatusCode::NOT_FOUND);
 
-    let res = client.get("/foo/?bar=baz").send().await;
-    assert_eq!(res.status(), StatusCode::PERMANENT_REDIRECT);
-    assert_eq!(res.headers().get("location").unwrap(), "/foo?bar=baz");
+    let res = client.get("/foo").send().await;
+    assert_eq!(res.status(), StatusCode::OK);
 }
 
 #[tokio::test]
-async fn without_trailing_slash() {
+async fn not_found_for_missing_trailing_slash() {
```

Comment:
I would argue given the `OK` test at the bottom, this test name is not accurate.

Reply:
I've removed the last assertion which might make things more clear. This test is for what happens when the request is missing a trailing slash.